File: blind_75/mergestringsalternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        resultString = ""
        length1, length2 = len(word1), len(word2)
        minLength = min(length1, length2)
        
        for i in range(minLength):
            resultString += word1[i] + word2[i]
        
        if length1 > minLength:
            resultString += word1[minLength:]
        elif length2 > minLength:
            resultString += word2[minLength:]
        
        return resultString

File: hogg_list/arrays_and_strings/mergeStringsAlternately.go
package main
func mergeAlternately(word1 string, word2 string) string {
    m := len(word1)
    n := len(word2)

    minLen := min(m, n)
    var result string 


    for i := 0; i < minLen; i++ {
        result += string(word1[i])
        result += string(word2[i])
    }

    if m > n {
        result += string(word1[minLen:])
    } else {
         result += string(word2[minLen:])
    }
    return result
}

func min(a, b int) int {
    if a < b {
        return a 
    }
    return b
}